src/common/s3.rs

```diff
@@ -12,6 +12,10 @@ impl ClientBuilder for S3ClientBuilder {
     }
 
     fn build(client: aws_smithy_client::Client, config: &aws_types::SdkConfig) -> Self::Client {
-        aws_sdk_s3::client::Client::with_config(client, config.into())
+        let config = aws_sdk_s3::config::Builder::from(config)
+            .force_path_style(true)
+            .build();
+
+        aws_sdk_s3::client::Client::with_config(client, config)
```

Comment:
This forces us into path style requests which localstack is more amenable to, theoretically [they support it](https://docs.localstack.cloud/user-guide/aws/s3/#getting-started) but I couldn't get it working in our integration test setup.

Virtual hosting is preferred, but the deprecation of path style has been delayed.

If we wanted to play around some more to get it working with localstack I would be happy to pair.